### PR TITLE
CRAFTERCMS-2812

### PR DESCRIPTION
**Pull request by dejan-brkic**
CMIS API v2 implemented

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2812

### CODE REVIEW

**Review by jdrossl — APPROVED** [review]

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -1456,6 +1461,270 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /cmis/list:
+    get:
+      tags:
+        - cmis
+      summary: List files and folders in a CMIS repository with an optional range for pagination
+      operationId: cmisList
+      description: 'Required permissions: "Read", "Write", "Create Content"'
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml]: isn't the permission `list_cmis`?

File: src/main/api/2/studio-api-2.yaml [hunk 2]
```diff
@@ -1456,6 +1461,270 @@ paths:
...
+        '503':
+          $ref: '#/components/responses/ServiceUnavailable'
+
+  /cmis/search:
+    get:
+      tags:
+        - cmis
+      summary: Search files and folders in a CMIS repository with an optional range for pagination
+      operationId: cmisSearch
+      description: 'Required permissions: "Read", "Write", "Create Content"'
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml]: isn't the permission `search_cmis`?

File: src/main/api/2/studio-api-2.yaml [hunk 3]
```diff
@@ -1456,6 +1461,270 @@ paths:
...
+        '503':
+          $ref: '#/components/responses/ServiceUnavailable'
+
+  /cmis/clone:
+    post:
+      tags:
+        - cmis
+      summary: Clone a file from a CMIS repository to Studio provided paths
+      operationId: cmisClone
+      description: 'Required permissions: "Read", "Write", "Create Content"'
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml]: isn't the permission `clone_content_cmis`?

File: src/main/api/2/studio-api-2.yaml [hunk 4]
```diff
@@ -1456,6 +1461,270 @@ paths:
...
+        '503':
+          $ref: '#/components/responses/ServiceUnavailable'
+
+  /cmis/upload:
+    post:
+      tags:
+        - cmis
+      summary:  Upload an asset file to CMIS repository
+      operationId: cmisUpload
+      description: 'Required permissions: "Read", "Write", "Create Content"'
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml]: isn't the permission `upload_content_cmis`?

**Review by avasquez614 — CHANGES_REQUESTED** [review]

**Review by avasquez614 — APPROVED** [review]

**Pull request merged**